sql-encryption-protector-byok/azuredeploy.json

```diff
@@ -0,0 +1,100 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "sqlServerName": {
+            "type": "string"
+        },
+        "sqlServerAdministrator": {
+        "type": "string"
+        },
+        "sqlServerAdministratorPwd": {
+            "type": "string"
+        },
+        "keyVaultName": {
+            "type": "string"
+        },
+        "keyName": {
+            "type": "string"
+        },
+        "keyVersion": {
+            "type": "string"
+        },
+        "location": {
+          "type": "string",
+          "defaultValue": "[resourceGroup().location]",
+          "metadata": {
+            "description": "Location for all resources."
+          }
+        }
+    },
+    "variables": {
+        "sqlServerKeyName": "[concat(parameters('keyVaultName'), '_', parameters('keyName'), '_', parameters('keyVersion'))]"
+    },
+    "resources": [
+    {
+      "type": "Microsoft.Sql/servers",
+      "apiVersion": "2015-05-01-preview",
+      "name": "[parameters('sqlServerName')]",
+      "location": "[parameters('location')]",
+      "identity": {
+        "type": "SystemAssigned"
+      },
+      "properties": {
+        "mode": "Incremental",
+        "administratorLogin": "[parameters('sqlServerAdministrator')]",
+        "administratorLoginPassword": "[parameters('sqlServerAdministratorPwd')]",
+        "version": "12.0"
+      },
+      "resources": []
+    },
+    {
+      "name": "[concat(parameters('keyVaultName'), '/add')]",
+      "type": "Microsoft.KeyVault/vaults/accessPolicies",
```

Comment:
since you're adding an access policy to the vault, you'll need to use a prereq template to provision a vault for this sample.

Reply:
Hello, thanks for your feedback: the whole point of this template was to actually rely on a global existing key vault (to bring your own existing key) - which was actually a customer case. I tried to take your feedback in consideration though, but it is not possible: by provisioning a Vault via Arm, I can only add a secret via Arm, (not a key), and SQL encryption protector can only be a key, not a secret. So I am stuck.

Reply:
ok, let's do this - add a PowerShell script to this same that will create a vault with the necessary perms and add the required key.  I'll run that on the CI subscriptions and we'll use that for the test.